src/providers/test.rs

```diff
@@ -71,9 +72,9 @@ fn build_mime_with_custom_headers() {
     assert_eq!("To: b@b.com", &message[1]);
     assert_eq!("Subject: subject", &message[2]);
     assert_eq!("MIME-Version: 1.0", &message[3]);
-    assert_eq!("x-foo: bar", &message[4]);
+    assert_eq!("X-Device-Id: baz", &message[4]);
```

Comment:
@philbooth this new test update for header capital case, could this break some other tests where we have something like `header['x-link']` instead of `X-Link`

Reply:
ok sounds good, there might be other external tests that might be affected (fxa pom), we just need to remember that this was the change.